pkg/issuer/acme/http/service.go

```diff
@@ -113,6 +112,15 @@ func buildService(crt *v1alpha1.Certificate, ch v1alpha1.ACMEOrderChallenge) *co
 			Selector: podLabels,
 		},
 	}
+
+	// http01 config is still optional, thus checking for presence and setting previous default
+	if issuer.GetSpec().ACME.HTTP01 == nil {
+		service.Spec.Type = corev1.ServiceTypeNodePort //TODO drop this line in future to use Kubernetes' default
+	} else if issuer.GetSpec().ACME.HTTP01.SolverServiceType != "" {
+		service.Spec.Type = issuer.GetSpec().ACME.HTTP01.SolverServiceType
+	}
```

Comment:
Can we refactor this `if .. else` into:

```
service.Spec.Type = corev1.ServiceTypeNodePort
if issuer.GetSpec().ACME.HTTP01 != nil && issuer.GetSpec().ACME.HTTP01.SolverServiceType != "" {
	service.Spec.Type = issuer.GetSpec().ACME.HTTP01.SolverServiceType
}
```

to follow go style conventions 😄

Reply:
... and also to handle the case where `issuer.GetSpec().ACME.HTTP01` is non-nil, and `issuer.GetSpec().ACME.HTTP01.SolverServiceType` is set to "".

The current patch would actually cause the 'default behaviour' to change to using ClusterIP (as we would not have set it to `ServiceTypeNodePort` anywhere)